activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -101,6 +101,13 @@ def _update_row(attribute_names, attempted_action = "update")
             attribute_names = attribute_names.dup if attribute_names.frozen?
             attribute_names << locking_column
 
+            if self[locking_column].nil?
+              raise(<<-MSG.squish)
```

Comment:
Raising `RuntimeError` (the default) isn't appropriate. This should raise one of Active Record errors. Not sure which one is best, but see `errors.rb` and other similar checks across Active Record to try to be consistent.

Reply:
Ok and thanks for the review! Would it make sense to change the error type raised by pessimistic locking too or the `RuntimeError` there is appropriate?

Reply:
Ah, I see. Yeah, it can be OK to raise a RuntimeError when something is considered to be a mistake of the caller, not a huge fan but seems consistent.